### ISSUE TITLE
[HUD] Improve MQ details

### ISSUE THREAD

**Issue by ianlandsman** [iron-io/issues#57]
The report for what's in the queue is pretty light right now. I think just a few more additions would make it super powerful. Primarily it'd be great to see the jobs in the queue. Is something stuck, is it taking a long time, etc.

I've attached an image of resque, something along those lines would be great.
![Screen 20Shot 202013-03-12 20at 2011 01 46 20AM](https://f.cloud.github.com/assets/53144/249876/1ce70b4c-8b3a-11e2-9b3a-eb89704078e4.png)

**Comment by carimura** [comment]
Just to be clear - you are using IronMQ as a job queue and each message/job is then processed by your own workers?

**Comment by ianlandsman** [comment]
Right, we're polling from our servers to pick things up off the queue. In other tools we've used there's been a bit more visibility into the specific jobs on the queue, that's really the core of what I think would be useful. We also use delayed jobs a bit and so being able to see those in there and know their state is useful.

It looks like we could potentially build our own with the API and peek, though I just looked quick and am not sure if there might be unintended consequences to that.

**Comment by carimura** [comment]
We've been thinking about this stuff.. we designed IronWorker as "workers as a service" which gives you all the standard job visibility, introspection, error reporting, etc, but your workers run on our servers. IronMQ is designed to be used for a variety of use cases, not just "jobs" so we can't _assume_ there's a status.

Are you using a job queue framework that we can look at (ie dj, resque, one for Laravel)?

**Comment by ianlandsman** [comment]
Hmm, perhaps I'm not being clear. When you're on this page:
https://hud.iron.io/mq/projects/XXXXX/queues/XX-name-XX

When Queue Size is > 0 it's be nice if you could click that number and see a list of all the items in the queue.

**Comment by carimura** [comment]
OK.. thought you were referring to specific things like "status" of jobs. But ya, this is also something we're thinking about also.

Thanks for the request and feedback. 

cc @thousandsofthem @treeder @edsrzf

**Comment by ianlandsman** [comment]
Ah ok. Yeah, I mean more status info could be useful like if a message has been picked but not deleted yet and stuff, how many times it's been tried, etc. We haven't used you guys long enough to see what happens when things go wrong so that could all be useful when things do.
